### PR TITLE
adding resource groups support

### PR DESCRIPTION
**Pull request by nakhanha**

### CODE REVIEW

File: presto-yarn-package/src/main/slider/package/templates/resource-groups.properties.j2 [hunk 1]
```diff
@@ -0,0 +1,2 @@
+resource-groups.configuration-manager=file
+resource-groups.config-file={{conf_dir}}/resource_groups.json
```
Comment (kokosing) [inline presto-yarn-package/src/main/slider/package/templates/resource-groups.properties.j2]: no new line at the end

File: presto-yarn-package/src/main/slider/package/templates/resource_groups.json.j2 [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      ]
+    }
+  ],
+  "selectors": [
+
+    {
+      "group": "global.adhoc_${USER}"
+    }
+  ]
+}
```
Comment (kokosing) [inline presto-yarn-package/src/main/slider/package/templates/resource_groups.json.j2]: no new line at then

File: presto-yarn-package/src/main/slider/package/templates/resource_groups.json.j2 [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+          "hardConcurrencyLimit": 2,
+          "maxQueued": 1,
+          "schedulingWeight": 9,
+          "schedulingPolicy": "query_priority"
+        }
+      ]
+    }
+  ],
+  "selectors": [
+
```
Comment (kokosing) [inline presto-yarn-package/src/main/slider/package/templates/resource_groups.json.j2]: remove extra blank line

File: presto-yarn-package/src/main/slider/package/templates/resource_groups.json.j2 [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+      "softMemoryLimit": "80%",
+      "hardConcurrencyLimit": 100,
+      "maxQueued": 1000,
+      "schedulingPolicy": "weighted",
+      "jmxExport": true,
+      "subGroups": [
+        {
+          "name": "adhoc_${USER}",
+          "softMemoryLimit": "10%",
+          "hardConcurrencyLimit": 2,
```
Comment (kokosing) [inline presto-yarn-package/src/main/slider/package/templates/resource_groups.json.j2:14]: how use can override these defaults? IIRC it is not possible. Do you think it is achievable to make it possible to for user to control these groups properties?

**Review by kokosing — COMMENTED** [review]
I would like to a see test for that in https://github.com/prestodb/presto-yarn/blob/master/presto-yarn-test/src/main/java/com/teradata/presto/yarn/test/PrestoClusterTest.java

File: presto-yarn-package/src/main/slider/package/scripts/configure.py [hunk 5]
```diff
@@ -38,8 +38,9 @@ def set_configuration(component=None):
```
Comment (kokosing) [inline presto-yarn-package/src/main/slider/package/scripts/configure.py:38]: Regarding commit message, please follow https://chris.beams.io/posts/git-commit/. Maybe `Add resource groups support`?

Comment (nakhanha) [reply presto-yarn-package/src/main/slider/package/scripts/configure.py:38]: Changed to "Add resource groups support"

File: presto-yarn-package/src/main/slider/package/templates/resource_groups.json.j2 (hunk 4 above)
Comment (nakhanha) [reply presto-yarn-package/src/main/slider/package/templates/resource_groups.json.j2:14]: I am not sure if it is possible. I have not seen any documentation for doing this. Meanwhile even query queues didnt support user config.

File: presto-yarn-package/src/main/slider/package/templates/resource-groups.properties.j2 (hunk 1 above)
Comment (nakhanha) [reply presto-yarn-package/src/main/slider/package/templates/resource-groups.properties.j2]: there is no new line at the end.

Comment (nakhanha) [reply presto-yarn-package/src/main/slider/package/templates/resource-groups.properties.j2]: added newline at end

File: presto-yarn-package/src/main/slider/package/templates/resource_groups.json.j2 (hunk 2 above)
Comment (nakhanha) [reply presto-yarn-package/src/main/slider/package/templates/resource_groups.json.j2]: added newline

File: presto-yarn-package/src/main/slider/package/templates/resource_groups.json.j2 (hunk 3 above)
Comment (nakhanha) [reply presto-yarn-package/src/main/slider/package/templates/resource_groups.json.j2]: removed

**Comment by kokosing** [comment]
@nakhanha Please ping me when it becomes ready for a review. Are you going to remove the support for query queues?

**Comment by nakhanha** [comment]
@kokosing , it should be ready to review,

I have already removed query queues template config in configure.py

There are no test for query queues. We can merge the changes now and I can work on tests later as I find some time

Thanks

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

File: presto-yarn-package/src/main/slider/package/scripts/configure.py [hunk 6]
```diff
@@ -38,8 +38,9 @@ def set_configuration(component=None):
 
     _template_config("{params.conf_dir}/config.properties", params, component)
     _template_config("{params.conf_dir}/node.properties", params)
-    _template_config("{params.conf_dir}/queues.json", params)
```
Comment (kokosing) [inline presto-yarn-package/src/main/slider/package/scripts/configure.py:41]: Please remove all the related files to query queues. Like https://github.com/prestodb/presto-yarn/blob/master/presto-yarn-package/src/main/slider/package/templates/queues.json.j2

File: presto-yarn-package/src/main/slider/package/scripts/params.py [hunk 7]
```diff
@@ -50,6 +50,7 @@
 coordinator_host = config['configurations']['global']['coordinator_host']
 presto_query_max_memory = config['configurations']['global']['presto_query_max_memory']
 presto_query_max_memory_per_node = config['configurations']['global']['presto_query_max_memory_per_node']
+presto_query_max_total_memory_per_node = config['configurations']['global']['presto_query_max_total_memory_per_node']
```
Comment (kokosing) [inline presto-yarn-package/src/main/slider/package/scripts/params.py]: this should be in separate commit, because it is not related to resource groups

**Review by kokosing — APPROVED** [review]
Please squash the commits, and please update documentation (if needed).

File: presto-yarn-package/src/main/slider/package/scripts/configure.py (hunk 6 above)
Comment (nakhanha) [reply presto-yarn-package/src/main/slider/package/scripts/configure.py:41]: removed

File: presto-yarn-package/src/main/slider/package/scripts/params.py (hunk 7 above)
Comment (nakhanha) [reply presto-yarn-package/src/main/slider/package/scripts/params.py]: removed from this PR. @kokosing  please merge

**Comment by kokosing** [comment]
Thank you, merged.
